server/src/main/java/game/snake/Snake.java

```diff
@@ -105,6 +108,16 @@ public void tick() {
 
         // update chunks
         currentChunk.append(encDirDelta, fast);
+
+        //TODO:
+        // - consider the snake head size when checking the snake head position
+        // - add a test for this
```

Comment:
Can we do that in this PR? A simple test with a snake moving to the next chunk and making sure its `GrowingSnakeChunk` is added to both should suffice IMO. Does not have to be a collision test.